tests/pruners_tests/test_threshold.py

```diff
@@ -56,10 +56,10 @@ def test_threshold_pruner_with_two_side() -> None:
 def test_threshold_pruner_with_invalid_inputs() -> None:
 
     with pytest.raises(TypeError):
-        optuna.pruners.ThresholdPruner(lower="0", upper=1.0)  # type: ignore
```

Comment:
I have noticed `float()` can take `"0"`.

Reply:
Thanks for the catch, you are correct. Accepting `"0"` seems reasonable and consistent.